parser/src/sema/scope.rs

```diff
@@ -0,0 +1,146 @@
+use std::{
+    borrow::Borrow,
+    collections::HashMap,
+    hash::Hash,
+    ops::{Index, IndexMut},
+};
+
+pub type Env<K, V> = Box<HashMap<K, V>>;
+
+#[derive(Clone)]
+pub enum LexicalScope<K, V> {
```

Comment:
nit: as a code style thing, we usually put headers above sections of code to separate things like constants, types, and the main code in the file, like the following:

```
// TYPES
//========================== (continued to a width of 100 chars)
```

This separation is useful for reading through files. It would be helpful especially to have some comments on the `LexicalScope`.

You can see examples in the [miden-vm repo](https://github.com/0xPolygonMiden/miden-vm/blob/main/processor/src/lib.rs)

Reply:
I think that can be helpful when there is a large number of top-level items in a single file, but I'm not sure how that applies here. This file defines a single type (`LexicalScope`) and a single type alias used to aid readability within the file, so having a pair of section comments to visually separate and identify them just seems like noise to me.

Of course I'll add them if you would prefer to have them rather than not, but it does feel out of place in Rust code IMHO. Generally you don't group declarations by type in Rust, but by domain. This is why you can define multiple `impl` blocks for a type (to group together declarations for a type which are used for a common purpose), and why you can inline declarations within a function body (e.g. constants, newtypes, etc.). In any case, let me know what convention you'd prefer I use in this repo, and I'll make a pass through and update all the modules I've touched so things are consistent.

Reply:
FYI - I'm going to merge this as-is into the `bitwalker/modules` branch, since both this PR and the next have been signed off, but just let me know about your preference here and I can apply a fixup to that branch instead

Reply:
This was really intended as more of a general note about the code style we've used as a team across other repositories, and I don't have a strong opinion about it here. I think the main reason I ended up noting it here is that there wasn't any comment on the `LexicalScope` struct, and we do in general try to comment most things.